cmd/podman/pod_ps.go

```diff
@@ -0,0 +1,574 @@
+	}
+	return
+}
+
+// generate the accurate header based on template given
+func (p *podPsTemplateParams) podHeaderMap() map[string]string {
+	v := reflect.Indirect(reflect.ValueOf(p))
+	values := make(map[string]string)
+
+	for i := 0; i < v.NumField(); i++ {
+		key := v.Type().Field(i).Name
+		value := key
+		if value == "ID" {
+			value = "Pod" + value
+		}
+		values[key] = strings.ToUpper(splitCamelCase(value))
+	}
+	return values
+}
+
+func sortPodPsOutput(sortBy string, psOutput podPsSorted) (podPsSorted, error) {
+	switch sortBy {
+	case "id":
+		sort.Sort(podPsSortedId{psOutput})
+	case "name":
+		sort.Sort(podPsSortedName{psOutput})
+	case "number":
+		sort.Sort(podPsSortedNumber{psOutput})
+	case "status":
+		sort.Sort(podPsSortedStatus{psOutput})
+	default:
+		return nil, errors.Errorf("invalid option for --sort, options are: id, names, or number")
+	}
+	return psOutput, nil
+}
+
+// getPodTemplateOutput returns the modified container information
+func getPodTemplateOutput(psParams []podPsJSONParams, opts podPsOptions) ([]podPsTemplateParams, error) {
+	var (
+		psOutput []podPsTemplateParams
+	)
+
+	for _, psParam := range psParams {
+		labels := formatLabels(psParam.Labels)
+		podID := psParam.ID
+		var ctrStr string
+
+		if !opts.NoTrunc {
+			podID = shortID(psParam.ID)
+			if opts.IdsOfContainers {
+				for i := 0; i < len(psParam.CtrsInfo); i++ {
+					psParam.CtrsInfo[i].Id = shortID(psParam.CtrsInfo[i].Id)
+				}
+			}
+		}
+		for _, ctrInfo := range psParam.CtrsInfo {
+			ctrStr += "[ "
+			if opts.IdsOfContainers {
+				ctrStr += ctrInfo.Id + " "
+			}
+			if opts.NamesOfContainers {
+				ctrStr += ctrInfo.Name + " "
+			}
+			if opts.StatusOfContainers {
+				ctrStr += ctrInfo.Status + " "
+			}
+			ctrStr += "] "
+		}
+		params := podPsTemplateParams{
+			ID:                 podID,
+			Name:               psParam.Name,
+			Status:             psParam.Status,
+			NumberOfContainers: psParam.NumberOfContainers,
+			Labels:             labels,
+			UsePodCgroup:       psParam.UsePodCgroup,
+			Cgroup:             psParam.Cgroup,
+			ContainerInfo:      ctrStr,
+		}
+
+		psOutput = append(psOutput, params)
+	}
+
+	return psOutput, nil
+}
+
+func getPodStatus(ctrs []*libpod.Container) (string, error) {
+	if len(ctrs) == 0 {
+		return CREATED, nil
+	}
+	statuses := map[string]int{
+		STOPPED: 0,
+		RUNNING: 0,
+		PAUSED:  0,
+		CREATED: 0,
+		DEAD:    0,
+	}
+	for _, ctr := range ctrs {
+		state, err := ctr.State()
+		if err != nil {
+			return "", err
+		}
+		switch state {
+		case libpod.ContainerStateStopped:
+			statuses[STOPPED]++
+		case libpod.ContainerStateRunning:
+			statuses[RUNNING]++
+		case libpod.ContainerStatePaused:
+			statuses[PAUSED]++
+		case libpod.ContainerStateCreated, libpod.ContainerStateConfigured:
+			statuses[CREATED]++
+		default:
+			statuses[DEAD]++
+		}
+	}
+
+	if statuses[RUNNING] > 0 {
+		return RUNNING, nil
+	} else if statuses[PAUSED] == len(ctrs) {
```

Comment:
nit: save `len(ctrs)` in a variable so we don't need to retrieve it again two lines below